app/src/main/java/com/kms/cura/view/RegDoc_PersonalFragment.java

```diff
@@ -0,0 +1,175 @@
+import java.util.Date;
+import java.util.List;
+
+/**
+ * A simple {@link Fragment} subclass.
+ */
+public class RegDoc_PersonalFragment extends Fragment implements View.OnClickListener {
+
+    Context context;
+    EditText edtFName, edtLName;
+    String FName, LName;
+    int sex;
+    int day, month, year;
+    Calendar calendar;
+    TextView tvBirth;
+    ImageButton btnChooseDate, btnNext;
+    Spinner spnSex;
+
+    public RegDoc_PersonalFragment() {
+        // Required empty public constructor
+        this.context = context;
+        calendar = Calendar.getInstance();
+        day = calendar.get(Calendar.DAY_OF_MONTH);
+        month = calendar.get(Calendar.MONTH) + 1;
+        year = calendar.get(Calendar.YEAR)-24;
+    }
+
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        // Inflate the layout for this fragment
+        View myFragmentView = inflater.inflate(R.layout.fragment_reg_doc__personal, container, false);
+        tvBirth = (TextView) myFragmentView.findViewById(R.id.tvBDate);
+        btnChooseDate = (ImageButton) myFragmentView.findViewById(R.id.btnChooseDate);
+        btnChooseDate.setOnClickListener(this);
+        btnNext = (ImageButton) myFragmentView.findViewById(R.id.btnRegisterNext);
+        btnNext.setOnClickListener(this);
+        edtFName = (EditText) myFragmentView.findViewById(R.id.edtDocReg_FirstName);
+        edtLName = (EditText) myFragmentView.findViewById(R.id.edtDocReg_LastName);
+        spnSex = (Spinner) myFragmentView.findViewById(R.id.spnSex);
+        // reform data if navigate from the next page
+        Bundle bundle = getArguments();
+        if (bundle != null) {
+            setDateString(bundle.getInt(RegisterDoctorActivity.DOB_DAY, day), bundle.getInt(RegisterDoctorActivity.DOB_MONTH, month), bundle.getInt(RegisterDoctorActivity.DOB_YEAR, year));
+            edtFName.setText(bundle.getString(RegisterDoctorActivity.FIRST_NAME, ""));
+            edtLName.setText(bundle.getString(RegisterDoctorActivity.LAST_NAME, ""));
+            String sex = bundle.getString(RegisterDoctorActivity.SEX, "");
+            if (sex.equals(RegisterDoctorActivity.SEX_MALE))
+                spnSex.setSelection(0);
+            else if (sex.equals(RegisterDoctorActivity.SEX_FEMALE))
+                spnSex.setSelection(1);
+        } else {
+            setDateString(day, month, year);
+        }
+        return myFragmentView;
+    }
+
+    @Override
+    public void onClick(View v) {
+        if (v.getId() == R.id.btnChooseDate) {
+            Dialog dateDialog = new DatePickerDialog(getActivity(),  AlertDialog.THEME_HOLO_LIGHT, myDateListener, year, month, day);
+            //newFragment.show(getFragmentManager(), "datePicker");
+            dateDialog.show();
+        } else if (v.getId() == R.id.btnRegisterNext) {
+            // check input
+            if (checkInput()) {
```

Comment:
In this, I want to separate which button is being clicked and which actions to be taken when that button is clicked, later we might add something to do when the checkInput() returns false